shared/src/types/ethereum_events.rs

```diff
@@ -33,20 +66,149 @@ pub struct TransferToNamada {
 }
 
 /// Represents Ethereum assets on the Ethereum blockchain
-#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, BorshSchema)]
+#[derive(
+    Debug, PartialEq, Eq, Clone, BorshSerialize, BorshDeserialize, BorshSchema,
+)]
 pub enum EthereumAsset {
-    /// Native ETH
-    Eth,
     /// An ERC20 token and the address of its contract
     ERC20(EthAddress),
 }
 
+/// A fraction of the total voting power. This should always be a reduced
+/// fraction that is between zero and one inclusive.
+#[derive(Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+pub struct FractionalVotingPower(Ratio<u64>);
+
+impl FractionalVotingPower {
+    /// Create a new FractionalVotingPower. It must be between zero and one
+    /// inclusive.
+    pub fn new(numer: u64, denom: u64) -> Result<Self> {
```

Comment:
I used `eyre` here instead of `thiserror` I wasn't sure when the caller would need to discriminate between an error when denominator is 0, and when the fraction is > 1, I think we will only care if there is an error or not